gems/kaminari-core/1.2/kaminari-core.rbs

```diff
@@ -0,0 +1,40 @@
+module Kaminari
+  def self.config: () -> Kaminari::Config
+  def self.configure: () { (Kaminari::Config) -> void } -> void
+
+  class Config
+    attr_accessor default_per_page: Integer
+    attr_accessor max_per_page: Integer?
+    attr_accessor window: Integer
+    attr_accessor outer_window: Integer
+    attr_accessor left: Integer
+    attr_accessor right: Integer
+    attr_accessor page_method_name: Symbol | String
+    attr_accessor max_pages: Integer?
+    attr_accessor params_on_first_page: bool
+
+    attr_writer param_name: Symbol | String | _Caller
+
+    def param_name: () -> (Symbol | String)
+  end
+
+  module PageScopeMethods[Relation] : _Model_Relation[Relation]
```

Comment:
Is `Relation` type parameter necessary?
`self` is appropriate for the returned type of `per`, `limit` and `offset` methods.